### ISSUE TITLE
False positive on vue/no-mutating-props

### ISSUE THREAD

**Issue by asselin** [vuejs/eslint-plugin-vue#2080]
**Checklist**

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.32.0
- **eslint-plugin-vue version:** 9.9.0
- **Node version:** 16.18.1
- **Operating System:** Mac

**Please show your full configuration:**

```json5
/* eslint-env node */
require('@rushstack/eslint-patch/modern-module-resolution');

module.exports = {
  root: true,
  ignorePatterns: ['coverage/*'],
  extends: [
    'plugin:vue/essential',
    'eslint:recommended',
    '@vue/eslint-config-typescript/recommended',
    '@vue/eslint-config-prettier',
    'plugin:json/recommended',
  ],
  rules: {
    '@typescript-eslint/no-non-null-assertion': 'off',
    '@typescript-eslint/no-unused-vars': ['error', { argsIgnorePattern: '^_' }],
  },
};
```

**What did you do?**	

```vue
<!-- ------------- Parent.vue -------------- -->
<template>
  <child-comp v-model="homeAddress" />
</template>

<script lang="ts">
import { defineComponent } from 'vue';
import ChildComp from './Child.vue';

export default defineComponent({
  name: 'ParentComp',
  components: {
    ChildComp,
  },
  data: () => ({
    homeAddress: { address1: '123 Main st' },
  }),
});
</script>

<!-- ------------- Child.vue -------------- -->
<template>
  <div>
    Enter street address
    <input v-model="value.address1" />
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue';

export default defineComponent({
  name: 'ChildComp',
  props: {
    value: {
      type: Object,
      required: true,
    },
  },
});
</script>

```

**What did you expect to happen?**
No ESLint errors.

**What actually happened?**
Received the error for Child.vue: "4:21  error  Unexpected mutation of "value" prop  vue/no-mutating-props".

Child does not mutate the prop, it mutates a property inside the passed prop.  Vue does not give any error at runtime about mutating a prop.  The test cases for this rule seem to indicate that this should be valid... but the rule is complaining.

**Repository to reproduce this issue**
(https://github.com/asselin/eslint-vue-bug)

**Comment by FloEdelmann** [comment]
Duplicate of #1371
